File: programers_Lv2/귤 고르기.js
function solution(k, tangerine) {
  let map = new Map();

  tangerine.forEach((v) => {
    if (map.has(v)) {
      map.set(v, map.get(v) + 1);
    } else {
      map.set(v, 1);
    }
  });

  const arr = [...map.values()];

  arr.sort((a, b) => a - b);

  let sum = (cnt = 0);
  while (true) {
    sum += arr.pop();
    cnt++;
    if (sum >= k) {
      return cnt;
    }
  }
}

console.log(solution(4, [1, 3, 2, 5, 4, 5, 2, 3])); // 2

File: programers_Lv2/귤 고르기.py
def solution(k, tangerine):
    d = dict()

    # 이중 for문 사용시 시간초과 발생하였음.
    for v in tangerine:
        if v not in d:
            d[v] = 1
        else:
            d[v] += 1
    
    lst = list(d.values())

    lst.sort()

    sum = 0
    cnt = 0
    while True:
        sum += lst.pop()
        cnt += 1
        if sum >= k:
            return cnt

# print(solution(6, [1, 3, 2, 5, 4, 5, 2, 3])) # 3
print(solution(4, [1, 3, 2, 5, 4, 5, 2, 3])) # 2